quantum/backlight/backlight_chibios.c

```diff
@@ -100,7 +109,7 @@ void backlight_set(uint8_t level) {
     } else {
         // Turn backlight on
         uint32_t duty = (uint32_t)(cie_lightness(rescale_limit_val(0xFFFF * (uint32_t)level / BACKLIGHT_LEVELS)));
-        pwmEnableChannel(&BACKLIGHT_PWM_DRIVER, BACKLIGHT_PWM_CHANNEL - 1, PWM_FRACTION_TO_WIDTH(&BACKLIGHT_PWM_DRIVER, 0xFFFF, duty));
+        pwmEnableChannel(&BACKLIGHT_PWM_DRIVER, BACKLIGHT_PWM_CHANNEL - 1, PWM_FRACTION_TO_WIDTH(&BACKLIGHT_PWM_DRIVER, BACKLIGHT_PWM_COUNTER_FREQUENCY, duty));
```

Comment:
This `0xFFFF` should not be changed — it is actually the maximum possible value of `duty`, which does not depend on the underlying PWM timer frequency.

`PWM_FRACTION_TO_WIDTH(pwmp, denominator, numerator)` actually calculates `pwmp->period * numerator / denominator`, which is the pulse width that would give the `numerator/denominator` duty fraction.

Reply:
You are right, I didn't fully understand the code here.